SALib/analyze/sobol.py

```diff
@@ -87,50 +88,23 @@ def analyze(problem, Y, calc_second_order=True, num_resamples=1000,
 def first_order(A, AB, B):
     # First order estimator following Saltelli et al. 2010 CPC, normalized by
     # sample variance
-    return np.mean(B * (AB - A)) / np.var(np.r_[A, B])
-
-
-def first_order_confidence(A, AB, B, num_resamples, conf_level):
-    s = np.empty(num_resamples)
-    for i in range(num_resamples):
-        r = np.random.randint(len(A), size=len(A))
-        s[i] = first_order(A[r], AB[r], B[r])
-
-    return norm.ppf(0.5 + conf_level / 2) * s.std(ddof=1)
+    return np.mean(B * (AB - A), axis=0) / np.var(np.r_[A, B], axis=0)
 
 
 def total_order(A, AB, B):
     # Total order estimator following Saltelli et al. 2010 CPC, normalized by
     # sample variance
-    return 0.5 * np.mean((A - AB) ** 2) / np.var(np.r_[A, B])
-
-
-def total_order_confidence(A, AB, B, num_resamples, conf_level):
-    s = np.empty(num_resamples)
-    for i in range(num_resamples):
-        r = np.random.randint(len(A), size=len(A))
-        s[i] = total_order(A[r], AB[r], B[r])
-
-    return norm.ppf(0.5 + conf_level / 2) * s.std(ddof=1)
+    return 0.5 * np.mean((A - AB) ** 2, axis=0) / np.var(np.r_[A, B], axis=0)

```

Comment:
And the option for total order:
![snippet](https://cloud.githubusercontent.com/assets/3727919/7533301/2d45a5b2-f568-11e4-9709-b8b834ca2480.png)

    1 - np.divide(np.dot(B, AB) - f2, np.dot(A, A) - f2)